scripts/Muon/GUI/Common/fitting_widgets/basic_fitting/workspace_selector_view.py

```diff
@@ -0,0 +1,96 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from mantidqt.utils.qt import load_ui
+
+from qtpy.QtWidgets import QWidget
+
+ui_workspace_selector, _ = load_ui(__file__, "workspace_selector.ui")
+
+
+class WorkspaceSelectorView(QWidget, ui_workspace_selector):
+    """
+    The WorkspaceSelectorView is the cyclic workspace selector combobox, and is used to choose the workspace that
+    is currently active in an interface.
+    """
+
+    def __init__(self, parent: QWidget = None):
+        """Initialize the WorkspaceSelectorView."""
+        super(WorkspaceSelectorView, self).__init__(parent)
+        self.setupUi(self)
+
+        self.increment_parameter_display_button.clicked.connect(self.increment_dataset_name_combo_box)
+        self.decrement_parameter_display_button.clicked.connect(self.decrement_dataset_name_combo_box)
+
+    def set_slot_for_dataset_changed(self, slot) -> None:
+        """Connect the slot for the display workspace combo box being changed."""
+        self.dataset_name_combo_box.currentIndexChanged.connect(slot)
+
+    def update_dataset_name_combo_box(self, dataset_names: list) -> None:
+        """Update the data in the parameter display combo box."""
+        old_name = self.dataset_name_combo_box.currentText()
+
+        self.update_dataset_names_combo_box(dataset_names)
+
+        new_index = self.dataset_name_combo_box.findText(old_name)
+        new_index = new_index if new_index != -1 else 0
+
+        self.dataset_name_combo_box.setCurrentIndex(new_index)
+        # This signal isn't always sent, so I will emit it manually.
```

Comment:
should we block signals before hand, so it will always be a single signal? Instead of sometimes getting a 1, 2 or 3?